src/database/models/problem-attachment.ts

```diff
@@ -0,0 +1,66 @@
+import { Model, DataTypes } from 'sequelize';
+import appSequelize from '../app-sequelize';
+import StudentGradeProblemAttachment from './student-grade-problem-attachment';
+
+interface ProblemAttachmentInterface {
+    id: number;
+    url: string;
+    userLocalFilename: string;
+    createdAt: Date;
+    updatedAt: Date;
+    active: boolean;
+}
+
+export default class ProblemAttachment extends Model implements ProblemAttachmentInterface {
+    public id!: number;
+    public url!: string;
+    public userLocalFilename!: string;
+    public active!: boolean;
+    
+    // timestamps!
+    public readonly createdAt!: Date;
+    public readonly updatedAt!: Date;
+
+    static constraints = {
+    }
+
+    static createAssociations(): void {
+        // This is a hack to add the associations later to avoid cyclic dependencies
+        /* eslint-disable @typescript-eslint/no-use-before-define */
+        ProblemAttachment.hasMany(StudentGradeProblemAttachment, {
+            as: 'studentGradeProblemAttachments',
+            foreignKey: 'problemAttachmentId',
+            sourceKey: 'id'
+        });
+
+        /* eslint-enable @typescript-eslint/no-use-before-define */
```

Comment:
do we also want a .hasMany(StudentGradeInstanceProblemAttachment?

Reply:
Oh yeah, I got distracted and forgot to implement the other ones

The belongs to is important because it creates the foreign keys, the hasMany is just for when you want to go in the opposite direction, so it's not a big deal if it's missing anway